sdk/tables/azure-data-tables/inc/azure/data/tables/tables_clients.hpp

```diff
@@ -374,31 +375,70 @@ namespace Azure { namespace Data { namespace Tables {
      * @param options Optional client options that define the transport pipeline policies for
      * authentication, retries, etc., that are applied to every request.
      */
-    explicit TableServicesClient(
+    explicit TableServiceClient(
         const std::string& serviceUrl,
         std::shared_ptr<Azure::Data::Tables::Credentials::SharedKeyCredential> credential,
         const TableClientOptions& options = {});
+
+    /**
+     * @brief Initializes a new instance of tableClient.
+     *
+     * @param serviceUrl A url referencing the table that includes the name of the account and the
+     * name of the table.
+     * @param credential The shared key credential used to sign requests.
+     * @param options Optional client options that define the transport pipeline policies for
+     * @param sasBuilder The shared access signature builder used to sign requests.
+     * authentication, retries, etc., that are applied to every request.
+     */
+    explicit TableServiceClient(
+        const std::string& serviceUrl,
+        std::shared_ptr<Azure::Data::Tables::Credentials::SharedKeyCredential> credential,
+        Azure::Data::Tables::Sas::AccountSasBuilder& sasBuilder,
+        const TableClientOptions& options = {});
+
     /**
      * @brief Initializes a new instance of tableClient.
      *
      * @param connectionString the connection string used to initialize.
      * @param options Optional client options that define the transport pipeline policies for
      * authentication, retries, etc., that are applied to every request.
-     * @return TableServicesClient.
+     * @return TableServiceClient.
      */
-    static TableServicesClient CreateFromConnectionString(
+    static TableServiceClient CreateFromConnectionString(
         const std::string& connectionString,
         const TableClientOptions& options = {});
 
     /**
-     * @brief List tables.
+     * @brief Create the table indicated in the tableName field of the client.
+     *
+     * @param context for canceling long running operations.
+     * @param tableName The name of the table to be created.
+     * @return Create table result.
+     */
+    Response<Models::Table> CreateTable(
+        std::string const& tableName,
+        Core::Context const& context = {});
+
+    /**
+     * @brief Delete the table indicated in the tableName field of the client.
+     *
+     * @param context for canceling long running operations.
+     * @param tableName The name of the table to be deleted.
+     * @return Delete table result.
+     */
+    Response<Models::DeleteResult> DeleteTable(
```

Comment:
Should "DeleteResult" be named "DeleteTableResult", or in this context, there is really nothing else you might delete, so "DeleteResult" would always unmistakingly be associated with tables?
I.e. there is no "DeleteTableRow()" method now or in the future, which would also return "DeleteTableRowResult"?